src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/LightCompiler.cs

```diff
@@ -1215,6 +1215,7 @@ private void EmitUnaryMethodCall(UnaryExpression node)
                 _instructions.EmitBranch(computed);
 
```

Comment:
Actually, I think this code must use the  EmitCoalescingBranch, just because of the stack behavior.